File: app/src/main/java/agstack/gramophone/utils/SharedPreferencesHelper.java
package agstack.gramophone.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by Abhisek on 04-03-2019.
 */

public class SharedPreferencesHelper {

    private static final String PREF_NAME = "sharedPreferences";
    private SharedPreferences.Editor editor;
    private SharedPreferences sharedPreferences;
    private static SharedPreferencesHelper sInstance;

    public SharedPreferencesHelper(Context context) {
        editor = context.getSharedPreferences(PREF_NAME, context.MODE_PRIVATE).edit();
        sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }


    public static synchronized void initializeInstance(Context context) {
        if (sInstance == null) {
            sInstance = new SharedPreferencesHelper(context);
        }
    }

    public static synchronized SharedPreferencesHelper getInstance() {
        if (sInstance == null) {
            throw new IllegalStateException(SharedPreferencesHelper.class.getSimpleName() +
                    " is not initialized, call initializeInstance(..) method first.");
        }
        return sInstance;
    }

        public void putString(String key, String value) {
            editor.putString(key,value);
            editor.apply();
        }
        public void putBoolean(String key, Boolean value) {
            editor.putBoolean(key, value);
            editor.apply();

        }
        public void putLong(String key, long value) {
            editor.putLong(key, value);
            editor.apply();
        }

        public void putFloat(String key, float value) {
            editor.putFloat(key, value);
            editor.apply();
        }

        public void removeId(String id) {
            editor.remove(id);
            editor.apply();
        }

        public String getString(String key) {return sharedPreferences.getString(key, null);}

        public boolean getBoolean(String key) {return sharedPreferences.getBoolean(key, false);}

        public long getLong(String key) {return  sharedPreferences.getLong(key, 0);}

        public Float getFloat(String key) {return  sharedPreferences.getFloat(key, 0);}
    public String getStringorDefault(String key,String defaultVal) {return sharedPreferences.getString(key, defaultVal);}



}

File: app/src/main/java/agstack/gramophone/utils/SharedPreferencesHelper.kt
package agstack.gramophone.utils

import android.content.SharedPreferences
import agstack.gramophone.utils.SharedPreferencesHelper
import android.content.Context
import kotlin.jvm.Synchronized

/**
 * Created by Abhisek on 04-03-2019.
 */

class SharedPreferencesHelper(context: Context) {
    private val editor: SharedPreferences.Editor
    private val sharedPreferences: SharedPreferences
    fun putString(key: String?, value: String?) {
        editor.putString(key, value)
        editor.apply()
    }

    fun putBoolean(key: String?, value: Boolean?) {
        editor.putBoolean(key, value!!)
        editor.apply()
    }

    fun putLong(key: String?, value: Long) {
        editor.putLong(key, value)
        editor.apply()
    }

    fun putFloat(key: String?, value: Float) {
        editor.putFloat(key, value)
        editor.apply()
    }

    fun removeId(id: String?) {
        editor.remove(id)
        editor.apply()
    }

    fun getString(key: String?): String? {
        return sharedPreferences.getString(key, "")
    }

    fun getBoolean(key: String?): Boolean {
        return sharedPreferences.getBoolean(key, false)
    }

    fun getLong(key: String?): Long {
        return sharedPreferences.getLong(key, 0)
    }

    fun getFloat(key: String?): Float {
        return sharedPreferences.getFloat(key, 0f)
    }

    fun getStringorDefault(key: String?, defaultVal: String?): String? {
        return sharedPreferences.getString(key, defaultVal)
    }

    companion object {
        private const val PREF_NAME = "sharedPreferences"
        private var sInstance: SharedPreferencesHelper? = null
        @JvmStatic
        @Synchronized
        fun initializeInstance(context: Context) {
            if (sInstance == null) {
                sInstance = SharedPreferencesHelper(context)
            }
        }

        @JvmStatic
        @get:Synchronized
        val instance: SharedPreferencesHelper?
            get() {
                checkNotNull(sInstance) {
                    SharedPreferencesHelper::class.java.simpleName +
                            " is not initialized, call initializeInstance(..) method first."
                }
                return sInstance
            }
    }

    init {
        editor = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE).edit()
        sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }
}